doc/neps/nep-0056-array-api-main-namespace.rst

```diff
@@ -0,0 +1,524 @@
+- JAX: enhancement proposal `Scope of JAX NumPy & SciPy Wrappers <https://jax.readthedocs.io/en/latest/jep/18137-numpy-scipy-scope.html#axis-2-array-api-alignment>`__
+  and `PR with initial implementation <https://github.com/google/jax/pull/16099>`__.
+
+
+Implementation
+--------------
+
+The tracking issue for Array API standard support
+(`gh-25076  <https://github.com/numpy/numpy/issues/25076>`__)
+records progress of implementing full support and links to related discussions.
+It lists all relevant PRs (merged and pending) that verify or provide array API
+support.
+
+As NEP 52 blends to some degree with this NEP, we can find some relevant implementations
+and discussion also on its tracking issue (`gh-23999 <https://github.com/numpy/numpy/issues/23999>`__).
+
+The PR that was merged as one of the first contained a new CI job that adds the
+`array-api-tests <https://github.com/data-apis/array-api-tests>`__ test suite.
+This way we had a better control over which batch of functions/aliases is being
+added each time, and be sure that the implementation conforms with the Array
+API standard (see `gh-25167 <https://github.com/numpy/numpy/pull/25167>`__).
+
+Then, we continued to merge one batch at the time, adding a specific API
+section. Below we list some of the more substantial ones, including some that
+we discussed in the previous sections of this NEP:
+
+- `gh-25167: MAINT: Add array-api-tests CI stage, add ndarray.__array_namespace__ <https://github.com/numpy/numpy/pull/25167>`__.
+- `gh-25088: API: Add Array API setops [Array API] <https://github.com/numpy/numpy/pull/25088>`__
+- `gh-25155: API: Add matrix_norm, vector_norm, vecdot and matrix_transpose [Array API] <https://github.com/numpy/numpy/pull/25155>`__
+- `gh-25080: API: Add and redefine numpy.bool [Array API] <https://github.com/numpy/numpy/pull/25080>`__
+- `gh-25054: API: Introduce np.isdtype function [Array API] <https://github.com/numpy/numpy/pull/25054>`__
+- `gh-25168: API: Introduce copy argument for np.asarray [Array API] <https://github.com/numpy/numpy/pull/25168>`__
+
+
+Alternatives
+------------
+
+The alternatives to implementing support for the array API standard in NumPy's
+main namespace include:
+
+- one or more of the superseded NEPs, or
+- making ``ndarray.__array_namespace__()`` return a hidden namespace with
+  compatible functions,
+- not implementing support for the array API standard at all.
+
+The superseded NEPs all have some drawbacks compared to the array API standard,
+and by now a lot of work has gone into the standard - as well as adoption by
+other key libraries. So those alternatives are not appealing. Given the amount
+of interest in this topic, doing nothing also is not appealing. The "hidden
+namespace" option would be a smaller change to this proposal. We prefer not to
+do that since it leads to duplicate implementations staying around, a more
```

Comment:
It's not impossible we may need it in the future, but it's not at all clear that we would ever need it. Versioning isn't needed as long as the added API surface and semantics in newer versions of the standard are a superset. Which should be true for the next version at least, and we should try hard to keep it that way. `special` extension isn't a given yet (it could be rejected, and certainly it isn't close to the finish line yet), and otherwise could probably be handled fairly easily in a module-level `__getattr__`.